ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+  
+  * Akka includes only transient timers at this point, reliable reminders need to be implemented using third-party extensions like Akka Quartz.
+
+### Runtime Implementation
+
+#### Overview
+
+Both systems consist of similar core components: Orleans’ Execution corresponds to akka-actor in that it is responsible for the actual execution of Actors, Messaging corresponds to akka-remote in that it manages TCP connections between the cluster nodes, and Hosting corresponds to akka-cluster plus ClusterSharding. One difference is that akka-actor also allows purely local message sends that are optimized to local reference passing whereas Orleans invocations are always sent via the Messaging component that consults with the Hosting component as to where the recipient currently resides. Akka’s lack of indirection allows substantial performance optimizations in this regard.
+
+#### Distributed Directory
+
+  * Orleans uses a one-hop distributed hash table that maps GUIDs to their current activations’ locations by way of partitioning the GUID hash space onto the cluster nodes. Local caches are used to optimize away the network round-trip in 90% of the cases.
+  
+  * Akka uses a sharding approach where the hash space of the logical Actor names is partitioned into shards of equal size and a ClusterSingleton coordinates the placement of these shards onto cluster nodes.
+
+While Orleans’ Hosting has the ability to relocate individual Actors, it must pay for this ability by a hash table that grows proportionally to the number of Grains used. Akka’s hash-based approach in contrast distributes the full location information (on an as-needed basis) because its size is bounded by the configurable number of shards, which allows it to scale further than Orleans in the number of Actors that are running.
+
+#### Strong Isolation
+
+  * Orleans ensures strong Grain isolation by copying all messages using the Serialization subsystem; application code can explicitly opt out of this behavior by using a marker class for wrapping invocation arguments.
+  
+  * Akka does not copy messages unless sending over a remote network link, trusting the user to not use mutable objects in Actor messages.
+  
+#### Asynchrony
+
+Both libraries implement a purely asynchronous programming model.
+
+#### Single-Threading
+
+Both libraries map Actors to Threads in an N:M fashion with N typically being much larger than M.
+
+#### Cooperative Multi-Tasking
+
+Both libraries allow Actors to process one message at a time without interruption or time limit.
+
+#### Serialization
+
+  * Orleans contains code generators for emitting the serialization and deserialization code for all Grain invocations.
+  
+  * Akka’s Serialization subsystem is fully configurable, users can use any library for mediating between objects and ByteString that they like.
```

Comment:
We recognize the lack of extensibility for Serialization as a limitation and are working on closing this gap: https://github.com/dotnet/orleans/issues/37. Just a matter of priorities.